sdks/python/apache_beam/testing/analyzers/github_issues_utils.py

```diff
@@ -0,0 +1,163 @@
+  logging.warning(
+      'A Github Personal Access token is required '
+      'to create Github Issues.')
+
+_BEAM_REPO_OWNER = 'AnandInguva'
+_BEAM_REPO_NAME = 'beam'
+_HEADERS = {
+    "Authorization": 'token {}'.format(_GITHUB_TOKEN),
+    "Accept": "application/vnd.github+json"
+}
+
+# Fill the GitHub issue description with the below variables.
+_ISSUE_DESCRIPTION_HEADER = """
+  Affected metric: `{}`
+"""
+_METRIC_INFO = "timestamp: {}, metric_value: `{}`"
+
+
+def create_or_comment_issue(
+    title: str,
+    description: str,
+    labels: Optional[List] = None,
+    issue_number: Optional[int] = None) -> Tuple[int, str]:
+  """
+  Create an issue with title, description with a label.
+  If an issue is already created and is open,
+  then comment on the issue instead of creating a duplicate issue.
+
+  Args:
+    title:  GitHub issue title.
+    description: GitHub issue description.
+    labels: Labels used to tag the GitHub issue.
+    issue_number: GitHub issue number used to find the already created issue.
+  """
+  if issue_number:
+    commented_on_issue, comment_url = comment_on_issue(
+      issue_number=issue_number,
+      comment_description=description)
+    if commented_on_issue:
+      return issue_number, comment_url
+
+  # Issue number was not provided or issue with provided number
+  # is closed. In that case, create a new issue.
+  url = "https://api.github.com/repos/{}/{}/issues".format(
+      _BEAM_REPO_OWNER, _BEAM_REPO_NAME)
+  data = {
+      'owner': _BEAM_REPO_OWNER,
+      'repo': _BEAM_REPO_NAME,
+      'title': title,
+      'body': description,
+  }
+  if labels:
+    data['labels'] = labels
+  response = requests.post(
+      url=url, data=json.dumps(data), headers=_HEADERS).json()
+  return response['number'], response['html_url']
+
+
+def comment_on_issue(issue_number: int,
+                     comment_description: str) -> Tuple[bool, Optional[str]]:
+  """
+  This method looks for an issue with provided issue_number. If an open
+  issue is found, comment on the open issue with provided description else
+  do nothing.
+
+  Args:
+    issue_number: A GitHub issue number.
+    comment_description: If an issue with issue_number is open,
+      then comment on the issue with the using comment_description.
+  """
+  url = 'https://api.github.com/repos/{}/{}/issues/{}'.format(
+      _BEAM_REPO_OWNER, _BEAM_REPO_NAME, issue_number)
+  open_issue_response = requests.get(
+      url,
+      json.dumps({
+          'owner': _BEAM_REPO_OWNER,
+          'repo': _BEAM_REPO_NAME,
+          'issue_number': issue_number
+      }),
+      headers=_HEADERS)
+  status_code = open_issue_response.status_code
+  open_issue_response = open_issue_response.json()
+  if status_code == 200 and open_issue_response['state'] == 'open':
+    data = {
+        'owner': _BEAM_REPO_OWNER,
+        'repo': _BEAM_REPO_NAME,
+        'body': comment_description,
+        issue_number: issue_number,
+    }
+    response = requests.post(
```

Comment:
one suggestion on the design also was to re-apply an `awaiting triage` label when we are bumping up an existing issue.